### PR TITLE
pluginConsolidate

### PR DESCRIPTION
**Pull request by kyr0**
This PR is based on PR #1619 and PR #1620 in this particular order :)

### CODE REVIEW

File: src/plugins/core/plugin_consolidate.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            ...options,
+            bundles: props.bundles,
+          });
+          props.fileContents = processedTemplate;
+        } catch (e) {
+          ctx.log.error(`Error processing the web-index template using consolidate: ${e.message}`);
+        }
+
+        return {
+          ...props,
```
Comment (nchanged) [inline src/plugins/core/plugin_consolidate.ts]: why spread? xD `return props`

File: src/plugins/core/plugin_consolidate.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import { IBundleWriteResponse } from '../../bundle/Bundle';
+import * as consolidate from 'consolidate';
+
+export function pluginConsolidate(engine: string, options: any) {
+  return async (ctx: Context) => {
+    ctx.ict.awaitOn(
+      'before_webindex_write',
+      async (props: { filePath: string; fileContents: string; bundles: Array<IBundleWriteResponse> }) => {
+        if (typeof consolidate[engine] !== 'function') {
+          ctx.log.error(`The template engine you selected is not available in consolidate: ${engine}`);
```
Comment (nchanged) [inline src/plugins/core/plugin_consolidate.ts]: how about ctx.fatal? I would make sense to stop the execution completely.

File: src/plugins/core/plugin_consolidate.ts (hunk 1 above)
Comment (kyr0) [reply src/plugins/core/plugin_consolidate.ts]: To clone it xD Because we can! 😂

File: src/plugins/core/plugin_consolidate.ts (hunk 2 above)
Comment (kyr0) [reply src/plugins/core/plugin_consolidate.ts]: I agree 👍

File: src/plugins/core/plugin_consolidate.ts (hunk 1 above)
Comment (nchanged) [reply src/plugins/core/plugin_consolidate.ts]: Nah don't clone them in fact I made a mistake in the first place, there is no need in returning anything

Comment (kyr0) [reply src/plugins/core/plugin_consolidate.ts]: Sry if it wasn't clear, but that was a joke ;) The clone was just something that remained from an old impl where I merged several objects

File: src/plugins/core/plugin_consolidate.ts (hunk 2 above)
Comment (kyr0) [reply src/plugins/core/plugin_consolidate.ts]: Done

File: src/plugins/core/plugin_consolidate.ts (hunk 1 above)
Comment (kyr0) [reply src/plugins/core/plugin_consolidate.ts]: Done

**Pull request merged**